docs/authenticate.js

```diff
@@ -50,35 +51,48 @@ function defaultErrorCallback(msg) {
   showError(msg, 'Authentication Error');
 }
 
-/**
- * Performs EarthEngine authentication and returns an auth object usable for
- * other things like GCS or Firebase.
- */
+/** Performs EarthEngine and Firebase authentication. */
 class Authenticator {
   /**
    * @constructor
-   * @param {Function} authCallback Will receive the auth response coming from
-   * authentication
    * @param {Function} eeInitializeCallback Called after EarthEngine
    *     initialization is complete
+   * @param {boolean} needsGdUser See {@link trackEeAndFirebase}
    */
-  constructor(authCallback, eeInitializeCallback) {
-    this.authCallback = authCallback;
+  constructor(eeInitializeCallback, needsGdUser) {
     this.eeInitializeCallback = eeInitializeCallback;
+    this.needsGdUser = needsGdUser;
     this.loginTasksToComplete = 2;
     this.gapiInitDone = new SettablePromise();
+    this.firebaseAuthFinished = new SettablePromise();
   }
 
-  /** Kicks off all processes. */
+  /**
+   * Kicks off all processes.
+   * @return {Promise<void>} Promise that completes when Firebase authentication
+   *     is finished
+   */
   start() {
     this.eeAuthenticate(() => this.onSignInFailedFirstTime());
     const gapiSettings = Object.assign({}, gapiTemplate);
     gapiSettings.scope = '';
     gapi.load('auth2', () => {
-      const initPromise = gapi.auth2.init(gapiSettings);
-      this.gapiInitDone.setPromise(initPromise);
-      initPromise.then(() => this.onLoginTaskCompleted());
+      this.gapiInitDone.setPromise(gapi.auth2.init(gapiSettings).then(() => {
+        const basicProfile =
+            gapi.auth2.getAuthInstance().currentUser.get().getBasicProfile();
+        if (this.needsGdUser &&
+            (!basicProfile || basicProfile.getEmail() !== gdUserEmail)) {
+          alert(
+              'You must be signed in as ' + gdUserEmail +
+              ' to access this page. Please open in an incognito window or ' +
+              'sign in as ' + gdUserEmail + ' here');
```

Comment:
maybe s/here/in this window (can you hyperlink a here?)

Reply:
Rephrased. I don't think we need to hyperlink, since it will happen automatically when the alert closes. The follow-up PR will have that issue :/